packages/experimental/NativeFontMetrics/ios/FRNFontMetrics.m

```diff
@@ -0,0 +1,140 @@
+#import "FRNFontMetrics.h"
+
+static NSDictionary<NSString *, NSNumber *> *FRNRecognizedTextStyles() {
+    static NSDictionary<NSString *, NSNumber *> *styles;
+    static dispatch_once_t onceToken;
+    dispatch_once(&onceToken, ^{
+        styles = @{
+            @"caption1": @(FRNTextStyleCaption1),
+            @"caption2": @(FRNTextStyleCaption2),
+            @"footnote": @(FRNTextStyleFootnote),
+            @"subheadline": @(FRNTextStyleSubheadline),
+            @"callout": @(FRNTextStyleCallout),
+            @"body": @(FRNTextStyleBody),
+            @"headline": @(FRNTextStyleHeadline),
+            @"title3": @(FRNTextStyleTitle3),
+            @"title2": @(FRNTextStyleTitle2),
+            @"title1": @(FRNTextStyleTitle1),
+            @"largeTitle": @(FRNTextStyleLargeTitle),
+        };
+    });
+    return styles;
+}
+
+@implementation RCTConvert (FRNTextStyle)
+
+RCT_ENUM_CONVERTER(FRNTextStyle, FRNRecognizedTextStyles(), FRNTextStyleUndefined, integerValue)
+
+@end
+
+NS_ASSUME_NONNULL_BEGIN
+
+UIFontMetrics *FRNUIFontMetricsForTextStyle(FRNTextStyle textStyle) {
+    static NSDictionary<NSNumber *, UIFontTextStyle> *mapping;
+    static dispatch_once_t onceToken;
+    dispatch_once(&onceToken, ^{
+        mapping = @{
+            @(FRNTextStyleCaption2): UIFontTextStyleCaption2,
+            @(FRNTextStyleCaption1): UIFontTextStyleCaption1,
+            @(FRNTextStyleFootnote): UIFontTextStyleFootnote,
+            @(FRNTextStyleSubheadline): UIFontTextStyleSubheadline,
+            @(FRNTextStyleCallout): UIFontTextStyleCallout,
+            @(FRNTextStyleBody): UIFontTextStyleBody,
+            @(FRNTextStyleHeadline): UIFontTextStyleHeadline,
+            @(FRNTextStyleTitle3): UIFontTextStyleTitle3,
+            @(FRNTextStyleTitle2): UIFontTextStyleTitle2,
+            @(FRNTextStyleTitle1): UIFontTextStyleTitle1,
+            @(FRNTextStyleLargeTitle): UIFontTextStyleLargeTitle,
+        };
+    });
+
+    id uiFontTextStyle = mapping[@(textStyle)] ?: UIFontTextStyleBody; // Default to body if we don't recognize the specified ramp
```

Comment:
Should try to avoid explicit `id` types in code -- this can just be `UIFontTextStyle`.